Documentation/guides/OneDotNetEmbeddedResources.md

```diff
@@ -0,0 +1,280 @@
+# [spec] Embedded Resources and `.nupkg` Files
+
+Traditionally, a Xamarin.Android class library can contain many types
+of Android-specific files:
+
+* `@(AndroidAsset)` files in `Assets\`
+* `@(AndroidEnvironment)` text files
+* `@(AndroidJavaLibrary)` `.jar` files
+* `@(AndroidResource)` files in `Resources\`
+* `@(EmbeddedNativeLibrary)` files in `lib\[arch]\*.so`
+
+These are packaged in different ways as `EmbeddedResource` files in
+the output assembly:
+
+* `__AndroidEnvironment__[filename]`
+  * `@(AndroidEnvironment)` files as-is
+* `__AndroidLibraryProjects__.zip`
+  * `assets` - `@(AndroidAsset)`
+  * `res` - `@(AndroidResource)`
+* `__AndroidNativeLibraries__.zip`
+  * `lib\[arch]*.so` - `@(EmbeddedNativeLibrary)`
+* `*.jar`
+  * `*.jar` files directly as `EmbeddedResource`
+
+The problem with this approach, is we have to inspect every assembly
+at build time and extract these files to a directory. Because we have
+a custom format that Android does not understand, we can't leave the
+files as-is. Android command-line tooling like `aapt2` works with
+files and directories on disk.
+
+Java binding projects have a slight variation:
+
+* `@(LibraryProjectZip)` can be `.aar` or `.zip` files and is
+  repackaged into `__AndroidLibraryProjects__.zip`. You can only have
+  one `@(LibraryProjectZip)` per project.
+* No support for `@(AndroidResource)` or `@(AndroidAsset)`.
+* `@(EmbeddedJar)` and `@(EmbeddedReferenceJar)` are packaged directly
+  as `EmbeddedResource` files.
+
+## New Approach in .NET 6
+
+We want to consider an approach that is NuGet-centric. When the above
+implementation was written, NuGet was just getting started. Developers
+still had the pattern of copying .NET assemblies from `bin` and
+committing it to their source control of choice. The single file
+approach worked well for this scenario.
+
+If we look at the general structure of a [NuGet package][nuget]:
+
+    lib
+        net6.0-android29\Foo.dll
+        net6.0-android30\Foo.dll
+    # Optional reference assemblies
+    ref
+        net6.0-android29\Foo.dll
+        net6.0-android30\Foo.dll
+    # Optional native libraries
+    runtimes
+        android-arm\native\libFoo.so
+        android-arm64\native\libFoo.so
+        android-x86\native\libFoo.so
+        android-x64\native\libFoo.so
+
+There is not a great place where all Android file types would fit
+following this pattern.
+
+In Android Studio, Android libraries are packaged as [`.aar`][aar]
+files. A Xamarin.Android library, `Foo.csproj`, could also generate an
+[`.aar`][aar] file in its `$(OutputPath)`:
+
+    Foo.aar
+        classes.jar
+        res/
+        assets/
+        libs/*.jar
+        jni/[arch]/*.so
+
+`@(AndroidEnvironment)` files are specific to Xamarin.Android, so we
+can make one addition to the file format:
+
+    Foo.aar
+        env/MyEnvironment.txt
+
+So the output of `Foo.csproj` would look like:
+
+    bin/Debug/[targetframework]/
```

Comment:
Kinda wondering what the convention is, if any, for `/` vs. `\` in paths.  I *thought* you were going for "`/` for `.zip` contents, `\` for on-disk contents," but here we have on-disk contents, and `/` is being used.

Should this instead use `\`?

Furthermore, instead of `[targetframework]` should this be `$(TargetFramework)`?